x/application/keeper/msg_server_delegate_to_gateway.go

```diff
@@ -3,15 +3,75 @@ package keeper
 import (
 	"context"
 
-	sdk "github.com/cosmos/cosmos-sdk/types"
 	"pocket/x/application/types"
+
+	sdkerrors "cosmossdk.io/errors"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 func (k msgServer) DelegateToGateway(goCtx context.Context, msg *types.MsgDelegateToGateway) (*types.MsgDelegateToGatewayResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// TODO: Handling the message
-	_ = ctx
+	logger := k.Logger(ctx).With("method", "DelegateToGateway")
+	logger.Info("About to delegate application to gateway with msg: %v", msg)
+
+	// DISCUSS_IN_THIS_PR: Should an application have to stake prior to delegation?
```

Comment:
@Olshansk WDYT here? I have it set up that the app must already be staked for it to delegate and I am confident this is the right way to do things. LMK if you think differently

Reply:
**tl;dr Yes.**

Fundamentally, it doesn't really matter because. If an app delegates without staking, it can't use relays anyhow. However, I just think things will be simpler to test/verify if it's a requirement so let's go with what you have.